shoop/core/cache/impl.py

```diff
@@ -72,7 +71,8 @@ def bump_version(self, cache_key):
         :type cache_key: str
         """
         namespace = _get_cache_key_namespace(cache_key)
-        version = str(time.time())
+        current_version = self.get_version(cache_key) or 1
+        version = current_version + 1
```

Comment:
This is also prone to race conditions, since this is not an atomic operation and multiple readers may be incrementing the version concurrently. Consider:

```
       Request 1       Request 2
       =========       =========
 T |   get_version
 i |    ==> 1
 m |                   get_version
 e |                    ==> 1      <-- oops, getting
   |                                   an old version while
 | |   increment       increment       Request 1 is already
 | |    ==> 2           ==> 2          bumping things...
 V |
   |   set_version
   |    ==> 2
   |                   set_version
   |                    ==> 2      <-- this should've been 3!
```

The correct fix is to either
* use a more high-resolution timer ([since Python 3.3, there're multiple high-res timers available on Unices](https://docs.python.org/3/library/time.html#time.clock_gettime)) if the current timer is the problem (as I understand from the commit message)
* or, more portably, to append a random string (Django's `get_random_string()` maybe?) to the version.